pkg/prom/agent.go

```diff
@@ -197,17 +292,31 @@ func (a *Agent) newInstance(c instance.Config) (instance.ManagedInstance, error)
 }
 
 func (a *Agent) validateInstance(c *instance.Config) error {
+	a.mut.Lock()
+	defer a.mut.Unlock()
+
 	return c.ApplyDefaults(&a.cfg.Global)
 }
 
 func (a *Agent) WireGRPC(s *grpc.Server) {
+	a.mut.Lock()
+	defer a.mut.Unlock()
+
 	if a.cfg.ServiceConfig.Enabled {
 		a.ha.WireGRPC(s)
 	}
```

Comment:
Ah, we need to handle this differently now. We'll need some kind of a deferred server that we can update.

Reply:
There's something similar for this in `WireAPI`, though that one is a little trickier.

Reply:
I'll create an `API` struct in the HA package that can be given a server.